use_cases/common/single_type/metrics/select_group_by.yaml

```diff
@@ -0,0 +1,66 @@
+title: Selection and Group By with Aggregation
+
+description: >
+  The most common query for metrics is to select a subset of time series, group
+  the series based on the unique values for a tag or dimension and aggregate the
+  results into a single value for each unit of time.
+
+inputs:
+  - metric series
+
+uses:
+  - type: Alerting
+    description: >
+      Commonly used when each group has similar behavior and the same static threshold
+      can apply to all groups.
+
+  - type: Visualization
+    description: >
+      Typically plotted in line graphs where patterns can be extracted to determine if
+      various groups of groups have similar behaviors or if one or more group exhibit
+      outlying behavior.
+
+examples:
+  - language: PromQL
+    query: >
+      sum by (job) (
+        http_requests_total{handler="/api/comments"}
+      )
+    description: >
+      Computes the sum of the metric `http_requests_total` where the tag `handler`
+      has the value `/api/comments` for every unique `job`.
+    references:
+      - https://prometheus.io/docs/prometheus/latest/querying/examples/#simple-time-series-selection
+      - https://prometheus.io/docs/prometheus/latest/querying/examples/#using-functions-operators-etc
+
+  - language: ASL
+    query: name,http_requests_total,:eq,handler,/api/comments,:eq,:and,:sum,(,job,),:by
+    description: >
+      Computes the sum of the metric `http_requests_total` where the tag `handler`
+      has the value `/api/comments` for every unique `job`.
+    references:
+      - https://netflix.github.io/atlas-docs/asl/ref/and/
+      - https://netflix.github.io/atlas-docs/asl/ref/eq/
+      - https://netflix.github.io/atlas-docs/asl/ref/by/
+
+  - language: ANSI SQL
```

Comment:
What is the table model you are assuming for this? A simple query like the one shown below would likely just have a single value instead of a time series like the other examples. Also, the time window would probably need to be explicit in the query where the others will usually get that from context outside the query expression.

Reply:
Good point. Most backends with a SQLish have query params for the step and time window. For a true ANSI SQL example we should spell out the pain of using a standard RBDMS.